packages/ii-login-button/src/ii-login-button.ts

```diff
@@ -0,0 +1,277 @@
+ *  // configure login and auth-client options here:
+ *  e.target.configure({
+ *    identityProvider: "http://localhost:3000"
+ *  });
+ * })
+ */
+export type IILoginButtonProps = {
+  createOptions?: AuthClientCreateOptions;
+  loginOptions?: AuthClientLoginOptions;
+  storage?: AuthClientStorage;
+};
+
+interface State {
+  authClient?: AuthClient;
+  isAuthenticated: boolean;
+  identity?: Identity;
+  ready: boolean;
+  createOptions?: AuthClientCreateOptions;
+  loginOptions?: AuthClientLoginOptions;
+  storage?: AuthClientStorage;
+}
+
+/**
+ * Internet Identity Login Button
+ * Drop-in web component to allow your users to log in with Internet Identity
+ *
+ * To use:
+ * Set up the component with <ii-login-button></ii-login-button>
+ * @example
+ * button.addEventListener('ready', (e)=>{
+ *  // configure login and auth-client options here:
+ *  e.target.configure(options);
+ * })
+ * button.addEventListener('login', (e)=>{
+ *  // Access authClient
+ *  const client = e.target.authClient;
+ *  // Principal
+ *  const principal = e.target.principal // also principalString
+ *  // AccountIdentifier for ICP ledger
+ *  const account = e.target.accountIdentifier
+ * })
+ */
+class IILoginButton extends LoginButton {
+  private _state: State;
+  constructor() {
+    super();
+    if (!this.shadowRoot) {
+      this.attachShadow({ mode: "open" });
+    }
+    this._state = {
+      isAuthenticated: false,
+      ready: false,
+    };
+
+    this.init();
+    this.render();
+  }
+
+  private init() {
+    AuthClient.create(this._state.createOptions)
+      .then(async (client) => {
+        this.setState({
+          ...this._state,
+          authClient: client,
+          isAuthenticated: await client.isAuthenticated(),
+          identity: await client.getIdentity(),
+        });
+        // fire ready event
+        if (!this._state.ready) {
+          const event = new ReadyEvent("ready");
+          this.dispatchEvent(event);
+          this.setState({ ready: true });
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+        this.setState({ authClient: undefined });
+      });
+  }
+
+  connectedCallback() {
+    const login = this._login.bind(this);
+    this.shadowRoot
+      ?.querySelector("#ii-login-button")
+      ?.addEventListener("click", login);
+  }
+
+  setState(newState: { [Property in keyof State]+?: State[Property] }) {
+    this._state = {
+      ...this._state,
+      ...newState,
+    };
+  }
+
+  private render() {
+    const shadowRoot = this.shadowRoot as ShadowRoot;
+    const oldButton = shadowRoot.querySelector("button");
+    const newButton = document.createElement("button");
+    newButton.id = "ii-login-button";
+
+    const logoRight = this.hasAttribute("logo-right");
+    const label = this.getAttribute("label") ?? "Login With Internet Identity";
+    if (logoRight) {
+      newButton.innerHTML = `
+          <span slot="label">${label}</span>
```

Comment:
slot can have default values I think, would that fits the label customization requirements and spare a config property?

```
<span><slot name="label">Login With Internent Identity</slot></span>
```

Reply:
Yeah just thinking that as a consumer I would probably have excepted to use the button as following

```
<ii-login-button>
    My custom text
</ii-login-button>
```

because that way the custom element behaves like an HTML element.

Reply:
I also like that syntax, but it's more overhead to observe changes to the `innerHTML` of a customElement than to hook into `attributeChanged`, since there's no native support for that for whatever reason

Reply:
Sure but why would this component need to hook on the changes in the `innerHTML`? The `slot` is projected even if it changes.